### DISCUSSION TITLE
New to vercel, don't know how to deploy a project

### DISCUSSION THREAD

**Discussion by ETwilight** [vercel/community#6383] [Help]
### Summary

In github pages, I run `npm run build` locally for my vue 3 project (with cli), and push the `/dist` folder onto github pages where then it will be deployed. But in vercel, it seems must to run a rebuild for me, can I skip this process? Currently, using the `gh-pages` branch causes build to fail which I have no idea why.

### Example

My github repo: [github.com/unrestrainedconcert.github.io](https://github.com/UnrestrainedConcert/unrestrainedconcert.github.io)

### Steps to Reproduce

Hi, I have an existing github pages with vue 3 and @vue/cli. 
In github, to deploy, I simply run `npm run build` locally and push the built version, which is in `/dist` folder, onto `gh-pages` branch where it will be automatically deployed since I set this branch for deployment.
However, in vercel, according to my previous tries, it seems that it must rebuild the project for me and I cannot use my built gh-pages branch. Is it really the case? Since there is some hidden contents in my master branch (which I managed to encrypt via pagecrypt), I don't really hope to push all my codes into the public git repo.

**Comment by amyegan** [comment]
Welcome, @ETwilight! There are a couple of options that might work for you.

**Option 1: Use [Vercel CLI](https://vercel.com/docs/cli) to deploy from a local build.** You control when a deployment is `preview` or `production`. And you get to keep a familiar setup. But you'll want to [turn auto-deployment off](https://vercel.com/docs/projects/project-configuration/git-configuration#git.deploymentenabled) for all branches. - https://vercel.com/docs/cli/deploying-from-cli#deploying-from-local-build-prebuilt

**Option 2: Skip your local `npm run build` step and let Vercel build it for you automatically.** You can use [environment variables](https://vercel.com/docs/projects/environment-variables) to store secret info. And you can set a [production branch](https://vercel.com/docs/deployments/git#production-branch) that is different from the main branch in your repo. - [vercel.com/docs/getting-started-with-vercel/import](https://vercel.com/docs/getting-started-with-vercel/import)

If you need further guidance with either of these options, please let me know!

**Reply by ETwilight** [reply]
Thank you! I will try that out!

**Comment by ETwilight** [comment]
@amyegan 
Hi! This is my approach:
Other framework, with override on everything
![image](https://github.com/vercel/community/assets/25543974/8e17dc30-df08-49eb-ab51-c11d44ccde0c)
Branch setting: gh-pages
![image](https://github.com/vercel/community/assets/25543974/4a617aa2-4bd0-49a4-8df3-f35974f6dadb)
My github repo: [github.com/unrestrainedconcert.github.io](https://github.com/UnrestrainedConcert/unrestrainedconcert.github.io)

But then, here still shows master branch which makes me confused
![image](https://github.com/vercel/community/assets/25543974/d250d804-45b1-40df-850a-c4d54f679e2d)
and when I click on it, it just build -- and then cancel the build.
![image](https://github.com/vercel/community/assets/25543974/7a99dcea-d945-47b1-b1c3-d7ea7ce23b86)
From here: it is also master branch, not gh-pages branch.
![image](https://github.com/vercel/community/assets/25543974/e0a679fc-1361-4409-90cd-42425d648fe9)

Where am I wrong with?
p.s. you can see my website via [here](https://unrestrainedconcert.github.io). This is a valid website, and gh-pages branch are generated by my local vue build.

**Comment by ETwilight** [comment]
I make another approach, which configures the project and change the directory of output to `/dist` folder. This leads to success, but I am still confused about why I can't do branch switching.
